File: backend/WedoIT.NextGen.Domain/DomainModels/Project.cs
namespace WedoIT.NextGen.Domain.DomainModels;

public class Project : BaseDomainModel
{
    public string Name { get; set; } 
    public Guid CreatorId { get; set; }
    public DateTime CreatedAt { get; set; }
    public string ClientName { get; set; } 
    public string RepositoryUrl { get; set; }
    public bool NeedHosting { get; set; }
    public bool HasDatabase { get; set; }
    public string Domain { get; set; } 
    public List<string> Types { get; set; } 
    public string Status { get; set; }
    public DateTime LastModified { get; set; } 


    public List<Block> Blocks { get; set; } = new List<Block>();

    public virtual Person Creator { get; set; } = null!;
    
    public virtual IEnumerable<History> Histories { get; set; } = new List<History>();
}

File: src/models/project.ts
import Block from "./block";
import Creator from "./creator";

type Project = {
  id: string;
  name: string;
  createdAt: Date;
  clientName: string;
  repositoryUrl: string;
  needHosting: boolean;
  hasDatabase: boolean;
  domain: string;
  type: string;
  lastModified: string;
  creator?: Creator | null;
  blocks: Block[];
};

export default Project;
